tests/experimental/datasources/test_utils.py

```diff
@@ -0,0 +1,377 @@
+            "value": 6,
+            "rendered_template": "$column maximum value must be greater than or equal to $min_value and less than or equal to $max_value.",
+        },
+        "expect_column_median_to_be_between": {
+            "value": 1,
+            "rendered_template": "$column median must be greater than or equal to $min_value and less than or equal to $max_value.",
+        },
+    }
+    assert len(validation_result.results) == 3
+
+    for r in validation_result.results:
+        assert r.success
+        assert (
+            r.result["observed_value"]
+            == expected_metric_values[r.expectation_config.expectation_type]["value"]
+        )
+
+        if include_rendered_content:
+            # There is a prescriptive atomic renderer on r.expectation_config
+            num_prescriptive_renderer = len(r.expectation_config.rendered_content)
+            assert (
+                num_prescriptive_renderer == 1
+            ), f"Expected exactly 1 rendered content, found {num_prescriptive_renderer}"
+            rendered_content = r.expectation_config.rendered_content[0]
+            assert rendered_content.name == AtomicPrescriptiveRendererType.SUMMARY
+            assert (
+                rendered_content.value.template
+                == expected_metric_values[r.expectation_config.expectation_type][
+                    "rendered_template"
+                ]
+            )
+
+            # There is a diagnostic atomic renderer on r, a validation result result.
+            num_diagnostic_render = len(r.rendered_content)
+            assert (
+                num_diagnostic_render == 1
+            ), f"Expected 1 diagnostic renderer, found {num_diagnostic_render}"
+            diagnostic_renderer = r.rendered_content[0]
+            assert (
+                diagnostic_renderer.name == AtomicDiagnosticRendererType.OBSERVED_VALUE
+            )
+            assert (
+                diagnostic_renderer.value.schema["type"]
+                == "com.superconductive.rendered.string"
+            )
+        else:
+            assert r.rendered_content is None
+            assert r.expectation_config.rendered_content is None
+
+    # Rudimentary test for data doc generation
+    docs_dict = context.build_data_docs()
+    assert "local_site" in docs_dict, "build_data_docs returned dictionary has changed"
+    assert docs_dict["local_site"].startswith(
+        "file://"
+    ), "build_data_docs returns file path in unexpected form"
+    path = docs_dict["local_site"][7:]
+    with open(path) as f:
+        data_doc_index = f.read()
+
+    # Checking for ge-success-icon tests the result table was generated and it was populated with a successful run.
+    assert "ge-success-icon" in data_doc_index
+    assert "ge-failed-icon" not in data_doc_index
+
+
+def run_data_assistant_and_checkpoint(
+    datasource_test_data: tuple[
+        AbstractDataContext, Datasource, DataAsset, BatchRequest
+    ]
```

Comment:
On second thought, I'm going to push for a namedtuple here - this is difficult to reason about as a len 4 tuple in my opinion

Reply:
@cdkini Again, this is not new.  We can update this implementation in a subsequent effort -- however, it is temporary anyway and will be replaced by a fuller file_path handling implementation soon.